### DISCUSSION TITLE
RxODE simulation with variance/covariance of off-diagonal OMEGA elements

### DISCUSSION THREAD

**Discussion by lirui0321** [nlmixr2/rxode2#668] [Q&A]
Hi Matt,

I have been reading clinical trial instructions you provided here (https://nlmixrdevelopment.github.io/RxODE/articles/RxODE-sim-var.html). In this example, you showed how cov matrix can be incorporated into the simulations. However, the cov matrix here only includes variance and covariance for theta and diagonal elements of omega matrix, but not off-diagonal elements of omega matrix. If I want to incorporate a cov matrix including variance/covariance for off-diagonal omega elements (for example, .cov file from NONMEM), is there a way to do that? The issue is that .cov file comes with off-diagonal names like OMEGA(2,8), but I assume that RxODE model does not know what that means, and I do not know how to pass that info to RxODE. I am assuming you probably had solved this issue when you built the nonmem2rx package, so wonder how you did it. 

Thank you!

Rui

**Comment by mattfidler** [comment]
Hi @lirui0321 

This is a methodology question, not a simulation issue.  Importing from `nonmem2rx` will use the same methodologies mentioned in the above post.  

The only method that uses correlations between the omega covariances and fixed effects is a multivariate normal simulation of the full problem.  That being said, it would have to simulate an `omega` matrix that is an invertible non-positive definite matrix.  In our (unpublished) tests with other systems, this often quite difficult to achieve.

The other issue with the multivariate normal distribution is there is no known relationship between it and the omega matrix (ie, means of items approach the normal distribution, but I don't think that that covariances approach the normal distribution). On the other hand, the inverse wishart is the conjugate prior of the covariance matrix.  Also the other simulation option (IJK matrix) is a established method for simulating a correlation matrix.

In the case of `nonmem2rx` these extra covariances are ignored just like they are in the case of a `nlmixr2` model.

If you find some methodology that integrates all of these without the problem of invertible non-positive matrices I would be interested in hearing about it.

**Reply by WXX155367** [reply]
Hi. Sorry to interrupt. I want to ask if it's true that if files in the .cov format can't be read in Non-MEM, then it can only read files in the .clt format, right? Can .clt files be read directly?

**Reply by mattfidler** [reply]
For nonmem2rx you can read in may sorts of formats.  You can read the control stream directly, you can read the xml and you can read the listing.  All files together will produce the best translation.

**Reply by WXX155367** [reply]
Could you give me an example of how to use nonmem2rx in lotri?

**Reply by mattfidler** [reply]
The best place to look is at the website: https://nlmixr2.github.io/nonmem2rx/

**Comment by lirui0321** [comment]
Many thanks for your feedback, Matt! When two fix effects are correlated, if I do not account for covariance (i.e., assuming two are independent with each other), then I may over-predict the uncertainty in the simulation. However, it is not intuitive to me how correlation between omega covariance and fixed effect (or correlation among omega covariance) affects the simulation. Since you just ignored these additional covariances in nonmem2rx, does it mean that the simulations would not be very sensitive to additional covariances? Or the reason for not including additional covariance is more of a technique issue like you outlined above?

**Reply by mattfidler** [reply]
The correlation between fixed effects and between subject variabilities are not accounted for.  In theory, the omega matrix is distributed by a different distribution than a multivariate normal and it is unclear what these correlations mean for this case.

There are people who account for these correlations, but they make other assumptions that may affect the suitability of your simulation (I just saw this today :smile:)
